### ISSUE TITLE
[dialogs] improve display of dialogs

### ISSUE THREAD

**Issue by vince-fugnitto** [eclipse-theia/theia#6905]
**Description**

Currently, **dialogs** which have a lot of content are not properly displayed (width-wise).

In order to fix their entire content, the dialog will take up as much width as possible which gives off a poor UX. Instead of placing the responsibility on developers to place carriage returns, the application should define proper `max-width` rules to enforce a consistent layout. 

Developers are still free to extend existing dialogs in order to style them as they so choose.

For example:

<div align='center'>

![image](https://user-images.githubusercontent.com/40359487/72611827-1810fc80-38f9-11ea-94b3-94793fd09ee3.png)

</div>

**Comment by vince-fugnitto** [comment]
We can add a new optional parameter to `DialogProps` for which we control the desired width of the dialog. This will mean that existing dialogs will not be broken by the change, but we can adjust internal dialogs with the new parameter.

https://github.com/eclipse-theia/theia/blob/8dcd4319b36f3acb9aa238b39cca10511a62251c/packages/core/src/browser/dialogs.ts#L24-L26

**Comment by JesterOrNot** [comment]
Can I work on this?

**Comment by vince-fugnitto** [comment]
> Can I work on this?

Sure! 👍

**Comment by JesterOrNot** [comment]
I've created the instance variable how do I implement the maxWidth part?

**Comment by vince-fugnitto** [comment]
> I've created the instance variable how do I implement the maxWidth part?

I was thinking of adding a new optional property to the `DialogProps`, and based on the value we'd update the styling of the dialog (`maxWidth`, `wordWrap`). Perhaps some changes will be need to be made to the `AbstractDialog` in order to be able to customize the outer container.

Then we can do something like the following in the constructor of a dialog:

```ts
this.container.setAttribute(`style`, `max-width: ${max-width} px`);
```

Callers could do the following:

```ts
new ConfirmDialog({
  title: 'title'
  maxWidth: 150,
  msg: 'msg',
  ok: 'yes',
  cancel: 'No'
});
```

**Comment by kaiyue0329** [comment]
@vince-fugnitto  Would using `flex-wrap: wrap`; on the dialog be an alternative solution?

**Comment by vince-fugnitto** [comment]
Closed by #7080
